### PR TITLE
Add urdf mesh converter

### PR DESCRIPTION
**Pull request by iory**
This package converts the colored output of the URDF Exporter plugin of SolidWorks into the format readable by ROS systems.
Specifically, this package generates colored `.dae` models from `.3dxml` files.

JSK users should refer to the URDF Exporter documentation for [guidance](https://docs.google.com/presentation/d/1c2fdw6f_ny4jizH5VS3M7yLNd60M3rzsZs-ACR1ghNI).

## Quick Start
After building this package, source your workspace. Then, simply execute the command below, specifying your .3dxml URDF file path and the desired output URDF path.

```
rosrun urdf_mesh_converter convert_urdf_mesh.py <YOUR_URDF_PATH> --output <OUTPUT_URDF_PATH>
```

For a practical example, you can apply this package to an actual URDF by running the following command:

```
rosrun urdf_mesh_converter convert_urdf_mesh.py $(rospack find urdf_mesh_converter)/samples/pitch_yaw_2axis_camera_module.urdf --output $(rospack find urdf_mesh_converter)/samples/pitch_yaw_2axis_camera_module_modified.urdf
```
image

@makabe0510 @pazeshun @Masanori-Konishi 
If you find this package necessary, I would greatly appreciate your review.

### CODE REVIEW

**Review by makabe0510 — APPROVED** [review]

**Comment by iory** [comment]
Please wait for reviews from others as well. 
This is not an urgent PR, so please take your time to review it when you have a moment.

**Comment by nasupium** [comment]
昨年度使用していたSolidworks2023版でも、今年度使用しているSolidworks2024版からもurdf-exporterを使ってrvizにロボットを表示することができました。

**Comment by pazeshun** [comment]
I confirmed `convert_urdf_mesh.py` can create colored URDF of my finger module:
Screenshot from 2024-06-17 19-42-58

SOLIDWORKS version: 2024 SP2.0
solidworks_urdf_exporter: Installed and used according to https://docs.google.com/presentation/d/1c2fdw6f_ny4jizH5VS3M7yLNd60M3rzsZs-ACR1ghNI/edit#slide=id.p
ROS version: Noetic

File: urdf_mesh_converter/samples/.gitignore
Comment (pazeshun) [inline urdf_mesh_converter/samples/.gitignore:3]: Duplicated?

File: urdf_mesh_converter/samples/.gitignore
Comment (pazeshun) [inline urdf_mesh_converter/samples/.gitignore:4]: Same comment as above.

File: urdf_mesh_converter/package.xml
Comment (pazeshun) [inline urdf_mesh_converter/package.xml:14]: Why is this not included in `ALPHABETICAL ORDER` lines?

File: urdf_mesh_converter/package.xml
Comment (pazeshun) [inline urdf_mesh_converter/package.xml:17]: Is `condition="$ROS_PYTHON_VERSION == 3"` not required? I think this enables us to install whole `jsk_model_tools` into python2 systems with peace of mind because `python3` will not be installed via `rosdep install`.

File: urdf_mesh_converter/package.xml
Comment (pazeshun) [inline urdf_mesh_converter/package.xml:21]: Same comment as above.

File: urdf_mesh_converter/CMakeLists.txt
Comment (pazeshun) [inline urdf_mesh_converter/CMakeLists.txt:27]: How about using `message(WARNING ...` and `return()` to keep installability of whole `jsk_model_tools` into python2 systems?

File: urdf_mesh_converter/CMakeLists.txt
Comment (pazeshun) [inline urdf_mesh_converter/CMakeLists.txt:10]: I think `message(WARNING ...` is easier to notice.

File: urdf_mesh_converter/README.md
Comment (pazeshun) [inline urdf_mesh_converter/README.md:3]: I think this sentence is a little bit misleading.
This package itself does not use URDF Exporter.
How about something like
```
This package converts the colored output of the URDF Exporter plugin of SolidWorks into the format readable by ROS systems.
Specifically, this package generates colored `.dae` models from `.3dxml` files.
```
?

**Review by pazeshun — CHANGES_REQUESTED** [review]

File: urdf_mesh_converter/samples/.gitignore
Comment (iory) [reply urdf_mesh_converter/samples/.gitignore]: Thanks. Fixed 
https://github.com/jsk-ros-pkg/jsk_model_tools/pull/255/commits/67460886373e5c66a5396fbc99a4b6c3f5fe278b

File: urdf_mesh_converter/samples/.gitignore
Comment (iory) [reply urdf_mesh_converter/samples/.gitignore]: Thanks. Fixed 
https://github.com/jsk-ros-pkg/jsk_model_tools/pull/255/commits/67460886373e5c66a5396fbc99a4b6c3f5fe278b

File: urdf_mesh_converter/package.xml
Comment (iory) [reply urdf_mesh_converter/package.xml]: Thanks.
https://github.com/jsk-ros-pkg/jsk_model_tools/pull/255/commits/d3833088b7baea0223bba869ec7dc364d549c70d
Update package.xml for conditional Python 3 dependency

- Change build_depend and exec_depend to conditionally depend on python3 if ROS_PYTHON_VERSION is 3.
- Correct buildtool_depend from catkin to catkin_virtualenv.

File: urdf_mesh_converter/package.xml
Comment (iory) [reply urdf_mesh_converter/package.xml]: Thanks.
https://github.com/jsk-ros-pkg/jsk_model_tools/commit/d3833088b7baea0223bba869ec7dc364d549c70d
Update package.xml for conditional Python 3 dependency

Change build_depend and exec_depend to conditionally depend on python3 if ROS_PYTHON_VERSION is 3.
Correct buildtool_depend from catkin to catkin_virtualenv.

File: urdf_mesh_converter/package.xml
Comment (iory) [reply urdf_mesh_converter/package.xml]: Fixed
https://github.com/jsk-ros-pkg/jsk_model_tools/commit/d3833088b7baea0223bba869ec7dc364d549c70d

File: urdf_mesh_converter/CMakeLists.txt
Comment (iory) [reply urdf_mesh_converter/CMakeLists.txt]: Thanks. Fixed.
https://github.com/jsk-ros-pkg/jsk_model_tools/pull/255/commits/b64b866662abb42d2622789f247704933559c595

File: urdf_mesh_converter/CMakeLists.txt
Comment (iory) [reply urdf_mesh_converter/CMakeLists.txt]: Thanks. Fixed.
https://github.com/jsk-ros-pkg/jsk_model_tools/pull/255/commits/b64b866662abb42d2622789f247704933559c595

File: urdf_mesh_converter/README.md
Comment (iory) [reply urdf_mesh_converter/README.md]: Thanks.
I changed your sentence because it is easier to understand.
https://github.com/jsk-ros-pkg/jsk_model_tools/pull/255/commits/c5b5ee03bf3485147673fa7e64fd3fc26227e553

**Comment by iory** [comment]
@pazeshun Thanks for your review!
I have modified the code based on your comment.

**Review by pazeshun — APPROVED** [review]

**Pull request merged**
